### PR TITLE
feat: [CDB-1195]: Dashboards AIDA Error Handling

### PR DESCRIPTION
**Pull request by Harness-JKhattak**
### Summary

- Added Error Handling to Dashboard's AIDA integration so that the user gets better feedback for successful/unsuccessful attempts at generating a Dashboard tile

#### Videos

**Success**

https://github.com/harness/harness-core-ui/assets/103499516/3037c013-e71b-418b-af4c-8bb3456128ab

**Failure**

https://github.com/harness/harness-core-ui/assets/103499516/3c95b5c2-5d3b-48fd-9a02-451c626c47e4

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [X] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by Harness-JKhattak** [comment]
retrigger jest

File: src/modules/45-dashboards/components/AidaChatRenderer/AidaChatRenderer.tsx [hunk 1]
```diff
@@ -25,12 +26,16 @@ const AidaChatRenderer: FC<AidaChatRendererProps> = ({ messages }) => {
   return (
     <Layout.Vertical spacing="large" className={css.chatRenderer}>
       {messages.map((message, i) => {
+        const isUser = message.role === MessageRole.User
         return (
-          <Container key={`aida-message-${i}`} className={css.userMsgWrapper}>
+          <Container
+            key={`aida-message-${i}`}
```
Comment (bgribben-harness) [inline src/modules/45-dashboards/components/AidaChatRenderer/AidaChatRenderer.tsx:32]: Map indexes aren't the best thing to use for generating a key, although I don't see anything more appropriate in this message object 😬

Comment (bgribben-harness) [reply src/modules/45-dashboards/components/AidaChatRenderer/AidaChatRenderer.tsx:32]: Nah bit much if it's a static list of shtuff 👍

File: src/modules/45-dashboards/strings/strings.en.yaml [hunk 2]
```diff
@@ -88,6 +88,9 @@ aida:
   helpText: Harness AI Development Assistant (AIDA) can help build Custom Dashboard widgets quickly and effectively.
   selectExplore: Select an explore for the widget
   selectVisualisation: Select a visualization for the widget
+  successGeneratingTile: AIDA successfully generated a Dashboard tile
+  failureGeneratingTile: AIDA dashboard tile generation failed
```
Comment (bgribben-harness) [inline src/modules/45-dashboards/strings/strings.en.yaml:92]: `Dashboard` vs `dashboard` - WHICH IS IT JK!?

**Review by bgribben-harness — APPROVED** [review]

**Comment by Harness-JKhattak** [comment]
retrigger jest

**Comment by Harness-JKhattak** [comment]
retrigger jest

**Review by sean-harness — APPROVED** [review]

**Review by bgribben-harness — APPROVED** [review]

**Pull request merged**